### PR TITLE
update links to other docs in index.html

### PR DESCRIPTION
**Pull request by IanMadd**
Signed-off-by: IanMadd <imaddaus@chef.io>

### Description

This updates the list of Chef documentation and their links on the index page. 

### Check List

- [ ] Spell Check
- [X] Local build
- [X] Examine the local build
- [ ] All tests pass

### CODE REVIEW

**Review by mjingle — APPROVED** [review]
tenor-209643958

File: chef_master/source/index.rst [hunk 1]
```diff
@@ -5,13 +5,15 @@ Site Map
 
 This is the documentation for:
 
-* Chef, including Chef Infra Server, Chef Infra Client, ChefDK, and related tools
-* Chef Automate
+* Chef Infra Server, Chef Infra Client, ChefDK, and related tools
+* Legacy Chef Automate 1
```
Comment (kagarmoe) [inline chef_master/source/index.rst]:
```suggestion
* Chef Automate 1.0 (deprecated)
```

File: chef_master/source/index.rst [hunk 2]
```diff
@@ -5,13 +5,15 @@ Site Map
...
 
-For information on Chef Habitat and Chef InSpec, see their respective documentation:
+For information on Chef Automate 2, Chef Habitat, Chef InSpec, and Chef Workstation, see their respective documentation:
```
Comment (kagarmoe) [inline chef_master/source/index.rst]:
```suggestion
Documentation for other Chef products: 
```

File: chef_master/source/index.rst [hunk 3]
```diff
@@ -5,13 +5,15 @@ Site Map
...
 
+* `Chef Automate 2 documentation <https://automate.chef.io/docs/quickstart/>`_
```
Comment (kagarmoe) [inline chef_master/source/index.rst]:
```suggestion
* `Chef Automate 2.0 <https://automate.chef.io/docs/quickstart/>`_
```

File: chef_master/source/index.rst [hunk 4]
```diff
@@ -5,13 +5,15 @@ Site Map
...
 * `Chef Habitat documentation <https://www.habitat.sh/docs/>`_
```
Comment (kagarmoe) [inline chef_master/source/index.rst]:
```suggestion
* `Chef Habitat <https://www.habitat.sh/docs/>`_
```

File: chef_master/source/index.rst [hunk 5]
```diff
@@ -5,13 +5,15 @@ Site Map
...
 * `Chef InSpec documentation <https://www.inspec.io/docs/>`_
```
Comment (kagarmoe) [inline chef_master/source/index.rst]:
```suggestion
* `Chef InSpec <https://www.inspec.io/docs/>`_
```

File: chef_master/source/index.rst [hunk 6]
```diff
@@ -5,13 +5,15 @@ Site Map
...
+* `Chef Workstation documentation <https://www.chef.sh/docs/chef-workstation/getting-started/>`_
```
Comment (kagarmoe) [inline chef_master/source/index.rst]:
```suggestion
* `Chef Workstation <https://www.chef.sh/docs/chef-workstation/getting-started/>`_
```

**Review by kagarmoe — CHANGES_REQUESTED** [review]

**Review by kagarmoe — APPROVED** [review]

**Pull request merged**